src/js/cards.js

```diff
@@ -247,28 +247,32 @@ function swipeNavigateCards(e) {
     }
 }
 
-function handleVideoTap(e) {
-    var $poster = $(e.currentTarget),
-        $video = $poster.closest('.flex').find('video');
+function prepareVideoHandling(cardHolder) {
+    var handleVideoTap = function(e) {
+        var $poster = $(e.currentTarget),
+            $video = $poster.closest('.flex').find('video');
 
-    $poster.hide();
-    $video.show();
-    $video.get(0).play();
+        $poster.hide();
+        $video.show();
+        $video.get(0).play();
 
-    $video.on('pause', function() {
-        $poster.show();
-        $video.hide();
-    });
+        $video.on('pause', function() {
+            $poster.show();
+            $video.hide();
+        });
 
-    $video.on('ended', function() {
-        $poster.show();
-        $video.hide();
-    });
+        $video.on('ended', function() {
+            $poster.show();
+            cardHolder.trigger('swipeLeft');
```

Comment:
Is it not required that we do `$video.hide()` to clean-up or reset it for the next play?